client/state/plugins/installed/selectors.js

```diff
@@ -124,11 +124,7 @@ function getPluginsSelector( state, siteIds, pluginFilter ) {
 
 export const getPlugins = createSelector(
 	getPluginsSelector,
-	( state, siteIds ) => [
-		state.plugins.installed.plugins,
-		isRequestingForAllSites( state ),
-		...siteIds.map( ( siteId ) => isRequesting( state, siteId ) ),
-	],
+	( state ) => state.plugins.installed.plugins,
```

Comment:
This prevents the `get plugins` from returning false when making the request, which allows fast navigation back and forth in the DataViews.

Reply:
Hi @paulopmt1, I think this change is causing an [issue I'm investigating now](https://github.com/Automattic/dotcom-forge/issues/10028).

AFAIK, listing the dependants here [adds the memoization](https://github.com/Automattic/wp-calypso/blob/9e3d6a5920c5af9485e263abbba849227b1019af/packages/state-utils/src/create-selector/index.ts#L109) within the `createSelector` function. Then the  `getPluginSelector` function returns the memoized value if there's one.

When we install plugins, we're [polling the plugins endpoint](https://github.com/Automattic/wp-calypso/blob/c8daea4cf6750cee0aaaccd03d3372a5024ccc22/client/my-sites/checkout/checkout-thank-you/marketplace/use-plugins-thank-you-data.tsx#L165-L167) to check if the newly installed plugin is listed there. This polling causes the `getPlugins` function to be re-run. After these changes, the selector _always_ returns the memoized values on subsequent calls, because the cache isn't being invalidated.

![CleanShot 2024-12-16 at 17 07 39@2x](https://github.com/user-attachments/assets/a9e670f9-ac89-4251-9f03-32d5ada25d62)

![CleanShot 2024-12-16 at 17 08 02@2x](https://github.com/user-attachments/assets/80fcf891-c121-452a-bcdf-cbd8463bd62b)

Could we perhaps implement an alternative way to support the fast navigation in DataViews, and restore this change?

Reply:
> Could we perhaps implement an alternative way to support the fast navigation in DataViews, and restore this change?

After this change we updated [the Plugins Manage](https://wordpress.com/plugins/manage) by removing the navigation and using a modal instead, so we can safely restore the old behavior here and it shouldn't affect that interface anymore.

Would you mind restoring that code @gcsecsey and adding me as a reviewer?

Reply:
That's great news, thanks so much @paulopmt1 for the context! 🙌 

> Would you mind restoring that code @gcsecsey and adding me as a reviewer?
Sure, I will restore the old version in [this PR](https://github.com/Automattic/wp-calypso/pull/97526), I added you as a reviewer just now. :)